sw/device/lib/runtime/pmp_regions.def

```diff
@@ -0,0 +1,46 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * \def PMP_REGION(region_id, config_reg_id)
+ *
+ * This is an X-Macro used for automatically deriving switch statements which
+ * link PMP region identifiers to associated information including their
+ * configuration register identifier.
```

Comment:
I'm not familiar with this pattern by the name "X-Macro" (though I *am* aware of this and similar `#include` tricks). Could you include a link that explains the pattern in detail, so that readers don't have to figure out this bit of preprocessor magic?

Reply:
I'm going to file an issue in the style guide about this, as a follow-up. I don't think this is the place to document it.

Reply:
Follow-up: #3854